### ISSUE TITLE
Start up error for job handler container only

### ISSUE THREAD

**Issue by pcm32** [galaxyproject/galaxy-helm#160]
Using the current ci values on minikube I'm getting this error on start only for the job handler deployment:

```
galaxy.jobs DEBUG 2020-08-30 11:24:14,481 Loading job configuration from /galaxy/server/config/job_conf.xml
galaxy.jobs ERROR 2020-08-30 11:24:14,481 Problem parsing the XML in file /galaxy/server/config/job_conf.xml, please correct the indicated portion of the file and restart Galaxy. '>=' not supported between instances of 'NoneType' and 'tuple'
Traceback (most recent call last):
  File "/galaxy/server/lib/galaxy/jobs/__init__.py", line 333, in __init__
    self._configure_from_dict(job_config_dict)
  File "/galaxy/server/lib/galaxy/jobs/__init__.py", line 372, in _configure_from_dict
    self._set_default_handler_assignment_methods()
  File "/galaxy/server/lib/galaxy/web_stack/handlers.py", line 148, in _set_default_handler_assignment_methods
    self.app.application_stack.init_job_handling(self)
  File "/galaxy/server/lib/galaxy/web_stack/__init__.py", line 117, in init_job_handling
    self._init_job_handler_assignment_methods(job_config, base_pool)
  File "/galaxy/server/lib/galaxy/web_stack/__init__.py", line 84, in _init_job_handler_assignment_methods
    self._set_default_job_handler_assignment_methods(job_config, base_pool)
  File "/galaxy/server/lib/galaxy/web_stack/__init__.py", line 537, in _set_default_job_handler_assignment_methods
    if ((dialect.name == 'postgresql' and dialect.server_version_info >= (9, 5))
TypeError: '>=' not supported between instances of 'NoneType' and 'tuple'
Failed to initialize Galaxy application
```

I think that I started seeing this error on rebase to the latest master.

**Comment by nuwang** [comment]
I've also run into this issue and I can't quite understand the root cause either. The quick fix is to have a null check:
```
if ((dialect.name == 'postgresql' and (dialect.server_version_info or (9, 5)) >= (9, 5))
```
However, that doesn't answer the question of why it's null in the first place - is it some sqlalchemy change? @natefoo or @mvdbeek?

**Comment by mvdbeek** [comment]
Don't know what changed, but `server_version_info` seems to only be available if a connection has been made previously.
```
diff --git a/lib/galaxy/web_stack/__init__.py b/lib/galaxy/web_stack/__init__.py
index 8f3a0a8d25..c371a14839 100644
--- a/lib/galaxy/web_stack/__init__.py
+++ b/lib/galaxy/web_stack/__init__.py
@@ -533,6 +533,9 @@ class WeblessApplicationStack(ApplicationStack):
         # isolation if it doesn't, or DB_PREASSIGN if the job_config doesn't allow either.
         conf_class_name = job_config.__class__.__name__
         remove_methods = [HANDLER_ASSIGNMENT_METHODS.DB_SELF]
+        with self.app.model.session.connection() as c:
+            # Force a connection so dialect.server_version_info is populated
+            pass
         dialect = self.app.model.session.bind.dialect
         if ((dialect.name == 'postgresql' and dialect.server_version_info >= (9, 5))
                 or (dialect.name == 'mysql' and dialect.server_version_info >= (8, 0, 1))):
```
works for me on dev. What Galaxy version did you see this with first ?

**Comment by mvdbeek** [comment]
(I do see this on 20.05, so I'm gonna target that)

**Comment by nuwang** [comment]
@mvdbeek Saw this first on 20.05. Thanks for the fix!

**Comment by pcm32** [comment]
Yes, this was working fine in 20.01, it started appearing when I changed the testing containers from 20.01 to 20.05. Thanks @mvdbeek !

**Comment by pcm32** [comment]
@nuwang @almahmoud do you know if a new container carrying the fix is now available? Currently the setup in the CI is still broken.

**Comment by pcm32** [comment]
Ok, the changes mentioned by @blankenberg are now available in pcm32/galaxy-k8s:20.05_7ddbf2 which hopefully someone will push to the galaxy user dockerhub account. But with that image this error doesn't occur.
